dll/win32/shell32/shell32.cpp

```diff
@@ -63,6 +63,53 @@ AddCommasW(DWORD lValue, LPWSTR lpNumber)
     return lpNumber;
 }
 
+// CreateEnvironmentBlock
+typedef BOOL (WINAPI *CREATEENVIRONMENTBLOCK)(LPVOID *, HANDLE, BOOL);
+
+/*
+ * Implemented
+ */
+EXTERN_C BOOL
+WINAPI
+RegenerateUserEnvironment(LPVOID *lpEnvironment, BOOL bUpdateSelf)
+{
+    HANDLE hUserToken;
+    OpenProcessToken(GetCurrentProcess(), TOKEN_READ | TOKEN_WRITE, &hUserToken);
+
+    HMODULE hUserEnv = GetModuleHandleA("userenv");
+
+    CREATEENVIRONMENTBLOCK pCreateEnvironmentBlock;
+    pCreateEnvironmentBlock = (CREATEENVIRONMENTBLOCK)GetProcAddress(hUserEnv, "CreateEnvironmentBlock");
```

Comment:
Why dynamically loading CreateEnvironmentBlock? userenv.lib is already statically linked to shell32 so there must no be any problem at all to use a direct call to this function.

Reply:
OK.